### PR TITLE
Changelog: Add missing entries for 3.3.4

### LINKED ISSUE #3624: Cobbler 3.3.4 release validation (2)
This is a follow-up to #3586. The purpose of this issue is to validate 700eb5b as a candidate for Cobbler 3.3.4.

The validation was performed in libvirt with openSUSE Tumbleweed 20240209 with 4 cores, 4GB RAM and 30GB disk space. mange_dns & manage_dhcp_v4 was enabled.

---

1. Install via repository (successful)
2. Turn on DHCP & DNS Management (successful)
3. `cobbler import` with Leap 15.5 (successful)
4. `cobbler sync` (successful)
5. `cobbler mkloaders` (successful)
6. `cobbler validate-autoinstalls` (successful)
7. `cobbler version` (partly successful)
8. `cobbler report`/`cobbler list` (successful)
9. `cobbler hardlink` (successful)
10. `cobbler check` (doesn't crash)
11. Restarting the daemon (successful)
12. Bootstrapping a VM via PXE (partly succeeded)
13. `cobbler status` (succeeded)

---

Release Blockers:

- Unattended PXE install: The default settings and DHCP don't match (#3623)

Problems:

- DNS: When using a subnet mask instead of an IP address Cobbler fails.
- `cobbler version` doesn't output the source hash due to a missing copy of the `.git` folder in the spec file.
- The DHCPD service refused to start due to in incorrect template by me.
- Unattended PXE install: The tree variable wasn't used for installation leading to the need to set `install=...` manually.
- Unattended PXE install: Default openSUSE AutoYaST template doesn't work without manually hitting continue multiple times.

### PR DESCRIPTION
**Pull request by SchoolGuy**
## Linked Items

Fixes #3624

## Description

This PR overhauls the changelog of Cobbler 3.3.4.

## Behaviour changes

Old: The Changelog was incomplete.

New: Towncrier generates a full changelog for 3.3.4

## Category

This is related to a:

- [ ] Bugfix
- [ ] Feature
- [ ] Packaging
- [ ] Docs
- [ ] Code Quality
- [ ] Refactoring
- [x] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [x] No tests required

### CODE REVIEW

File: changelog.d/3592.fixed
Comment (tiltingpenguin) [inline changelog.d/3592.fixed:1]: This changelog shouldn't be empty

File: changelog.d/3603.added
Comment (tiltingpenguin) [inline changelog.d/3603.added:1]: This changelog entry is associated with issue 3589. Also the PR already has a changelog entry included

File: changelog.d/3567.fixed
Comment (tiltingpenguin) [inline changelog.d/3567.fixed:1]: The associated issue is 3524

File: changelog.d/3477.fixed
Comment (tiltingpenguin) [inline changelog.d/3477.fixed:1]: The associated issue is 3473

File: changelog.d/3450.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+CLI: Fixed and error that mangled the types of certain flags for Cobbler items
```
Comment (tiltingpenguin) [inline changelog.d/3450.fixed:1]:
```suggestion
CLI: Fixed an error that mangled the types of certain flags for Cobbler items
```

**Review by tiltingpenguin — CHANGES_REQUESTED** [review]
Imo if there is no direct issue associated with a backport PR we should name the changelog after the original issue instead of the PR

File: changelog.d/3592.fixed
Comment (SchoolGuy) [reply changelog.d/3592.fixed:1]: The file is deleted and as such I figure that it would be fine?

**Review by tiltingpenguin — APPROVED** [review]
LGTM

**Comment by btravouillon** [comment]
Thank you so much Enno (@SchoolGuy)! I was relunctant to ask for a new release because I know you already have a lot on your plate. This is much appreciated you went ahead with 3.3.4!

**Pull request merged**